### PR TITLE
JDO-823: Fix sonarcloud issues of type Code Smells

### PR DESCRIPTION
**Pull request by mboapache**

### CODE REVIEW

File: api/src/main/java/javax/jdo/Enhancer.java [hunk 1]
```diff
@@ -155,19 +160,19 @@ private void run(String[] args) {
       properties = enhancer.getProperties();
       addVerboseMessage("MSG_EnhancerClass", enhancer.getClass().getName()); // NOI18N
       addVerboseMessage(
-          "MSG_EnhancerProperty",
+          MSG_ENHANCER_PROPERTY,
           PROPERTY_ENHANCER_VENDOR_NAME, // NOI18N
```
Comment (tzaeschke) [inline api/src/main/java/javax/jdo/Enhancer.java:164]: I think we can remove // NOI18N

File: api/src/main/java/javax/jdo/Enhancer.java [hunk 2]
```diff
@@ -85,6 +85,11 @@ public class Enhancer {
   /** Class file suffix */
   private static final String CLASS_FILE_SUFFIX = ".class"; // NOI18N
 
+  private static final String MSG_ENHANCER_CLASS_PATH = "MSG_EnhancerClassPath";
```
Comment (tzaeschke) [inline api/src/main/java/javax/jdo/Enhancer.java:88]: DO we need to add `// NOI18N` ? Is that still a thing?

File: api/src/main/java/javax/jdo/JDOHelper.java [hunk 3]
```diff
@@ -1119,6 +1119,7 @@ protected static Map<Object, Object> loadPropertiesFromResource(
         try {
           in.close();
         } catch (IOException ioe) {
+          // this code block is deliberately left empty,
```
Comment (tzaeschke) [inline api/src/main/java/javax/jdo/JDOHelper.java:1120]: Remove `,`?

File: api/src/main/java/javax/jdo/spi/JDOImplHelper.java [hunk 4]
```diff
@@ -1065,7 +1065,7 @@ public void nonBinaryCompatibleMakeDirty(Object pc, String fieldName) {
       try {
         if (si.makeDirty(pc, fieldName)) return;
       } catch (Throwable t) {
```
Comment (tzaeschke) [inline api/src/main/java/javax/jdo/spi/JDOImplHelper.java:1067]: Out of scope for this PR but maybe a case for a separate PR?
Maybe we should not catch/ignore `Throwable` here, that would include `OutOfMemoryException` and such.

File: api/src/main/java/javax/jdo/Enhancer.java (hunk 1 above)
Comment (mboapache) [reply api/src/main/java/javax/jdo/Enhancer.java:164]: Yes

File: api/src/main/java/javax/jdo/Enhancer.java (hunk 2 above)
Comment (mboapache) [reply api/src/main/java/javax/jdo/Enhancer.java:88]: Added NOI18N, also at a couple of other places.

File: api/src/main/java/javax/jdo/JDOHelper.java (hunk 3 above)
Comment (mboapache) [reply api/src/main/java/javax/jdo/JDOHelper.java:1120]: yes

File: api/src/main/java/javax/jdo/spi/JDOImplHelper.java (hunk 4 above)
Comment (mboapache) [reply api/src/main/java/javax/jdo/spi/JDOImplHelper.java:1067]: Please file a separate JIRA for this. Then we can discuss which exceptions should be swallowed.

Comment (tzaeschke) [reply api/src/main/java/javax/jdo/spi/JDOImplHelper.java:1067]: Done, see https://issues.apache.org/jira/browse/JDO-824.

**Pull request merged**